### PR TITLE
Update component docs (Header, Radio, Checkbox, Heading, Label, LabelGroup)

### PR DESCRIPTION
**Pull request by rezrah**
Part of #1701

### Previews
[Header](https://primer-react-l5jf5xzed-primer.vercel.app/react/Header)
[Radio](https://primer-react-l5jf5xzed-primer.vercel.app/react/Radio)
[Checkbox](https://primer-react-l5jf5xzed-primer.vercel.app/react/Checkbox)
[Heading](https://primer-react-l5jf5xzed-primer.vercel.app/react/Heading)
[Label](https://primer-react-l5jf5xzed-primer.vercel.app/react/Label)
[LabelGroup](https://primer-react-l5jf5xzed-primer.vercel.app/react/LabelGroup)

### CODE REVIEW

File: docs/content/Checkbox.mdx
Comment (colebemis) [inline docs/content/Checkbox.mdx:114]: Does this have a default value?

File: docs/content/Checkbox.mdx
Comment (colebemis) [inline docs/content/Checkbox.mdx:109]: Does this have a default value?

File: docs/content/Checkbox.mdx
Comment (colebemis) [inline docs/content/Checkbox.mdx:104]:
```suggestion
    type="(event: ChangeEvent) => void"
```

File: docs/content/Checkbox.mdx
Comment (colebemis) [inline docs/content/Checkbox.mdx:127]: `isPolymorphic` defaults to false
```suggestion
```

File: docs/content/Checkbox.mdx
Comment (colebemis) [inline docs/content/Checkbox.mdx:91]: The heading of prop tables is usually the name of the component: 

```suggestion
### Checkbox

<PropsTable>
```

File: docs/content/Checkbox.mdx
Comment (colebemis) [inline docs/content/Checkbox.mdx:138]:
```suggestion
## Status
```

File: docs/content/Header.mdx
Comment (colebemis) [inline docs/content/Header.mdx:59]: We don't use `` for these headings in any other places. We should probably be consistent here.
```suggestion
### Header
```

File: docs/content/Header.mdx
Comment (colebemis) [inline docs/content/Header.mdx:65]:
```suggestion
### Header.Item
```

File: docs/content/Header.mdx
Comment (colebemis) [inline docs/content/Header.mdx:72]:
```suggestion
### Header.Link
```

File: docs/content/Header.mdx
Comment (colebemis) [inline docs/content/Header.mdx:77]: Does this component also pass props down to the `<a>` element?

File: docs/content/Header.mdx
Comment (colebemis) [inline docs/content/Header.mdx:78]: Can we add a component checklist to this page?

File: docs/content/Label.mdx
Comment (colebemis) [inline docs/content/Label.mdx:46]:
```suggestion
  <PropsTableRow name="variant" type="'small' | 'medium' | 'large' | 'xl'" defaultValue="'medium'" />
```

File: docs/content/Radio.mdx
Comment (colebemis) [inline docs/content/Radio.mdx:91]:
```suggestion
## Status
```

File: docs/content/Radio.mdx
Comment (colebemis) [inline docs/content/Radio.mdx:81]:
```suggestion
```

File: docs/content/Radio.mdx
Comment (colebemis) [inline docs/content/Radio.mdx:71]:
```suggestion
    type="(event: ChangeEvent) => void"
```

File: docs/content/Heading.mdx
Comment (colebemis) [inline docs/content/Heading.mdx:4]: Can we add a link to the source here?

File: docs/content/Label.mdx
Comment (colebemis) [inline docs/content/Label.mdx:49]: Can we add a status checklist here?

File: docs/content/Header.mdx
Comment (colebemis) [inline docs/content/Header.mdx:8]: Should we nest the following headings under this? So the hierarchy would be:

```
# Examples

## With full-size item

...

## Props
```

File: docs/content/Checkbox.mdx
Comment (rezrah) [reply docs/content/Checkbox.mdx]: Updated

File: docs/content/Header.mdx
Comment (rezrah) [reply docs/content/Header.mdx]: Swapped it back, but it feels pretty ugly as the other headings include object syntax

File: docs/content/Header.mdx
Comment (rezrah) [reply docs/content/Header.mdx]: Swapped it back, but it feels pretty ugly as the other headings include object syntax.

File: docs/content/Checkbox.mdx
Comment (rezrah) [reply docs/content/Checkbox.mdx:109]: It's an optional prop, so no. Or are you setting undefined explicitly?

File: docs/content/Checkbox.mdx
Comment (rezrah) [reply docs/content/Checkbox.mdx:91]: The example heading in the Issue description [here](https://github.com/primer/react/issues/1701) shows it as Props, which feels sensible given the content it describes... or am I misunderstanding this? What was the reasoning behind repeating the component name here?

File: docs/content/Header.mdx
Comment (rezrah) [reply docs/content/Header.mdx]: Updated. But I do think the syntax-like content makes this quite an ugly heading

File: docs/content/Header.mdx
Comment (rezrah) [reply docs/content/Header.mdx]: Looks like it does, updated with base props. Is that more suitable? 🤔

File: docs/content/Heading.mdx
Comment (rezrah) [reply docs/content/Heading.mdx:5]: Done

File: docs/content/Label.mdx
Comment (rezrah) [reply docs/content/Label.mdx:49]: Added below 👇

File: docs/content/Label.mdx
Comment (rezrah) [inline docs/content/Label.mdx:56]: This component seems to expose `styled-system`, which I presume is an unnecessary dep?

File: docs/content/Label.mdx
Comment (rezrah) [inline docs/content/Label.mdx:59]: 75% on the some conditional behaviour

File: docs/content/Label.mdx
Comment (rezrah) [inline docs/content/Label.mdx:58]: Unsure about this one. Arguably it _could_ have some responsive behaviour to ensure size variants respect the constraints of the current screen and medium, so that the user doesn't have to do this themselves.

File: docs/content/Heading.mdx
Comment (rezrah) [reply docs/content/Heading.mdx:23]: Done. Can you please verify that you agree with the criteria @colebemis.

File: docs/content/LabelGroup.mdx
Comment (rezrah) [reply docs/content/LabelGroup.mdx:22]: Done. Can you please verify that you agree with the criteria @colebemis.

File: docs/content/Header.mdx
Comment (rezrah) [reply docs/content/Header.mdx]: Update: I've put a guard on `refType` so that row doesn't appear.

File: docs/src/props-table.js
Comment (rezrah) [inline docs/src/props-table.js:81]: guarding this so that it doesn't render this row unless a passed as a param. I think this is useful when the underlying component doesn't support refForwarding. cc @colebemis

File: docs/content/Checkbox.mdx
Comment (colebemis) [reply docs/content/Checkbox.mdx:109]: Would it make sense to say this defaults to`false`?

Comment (rezrah) [reply docs/content/Checkbox.mdx:109]: I think that's misrepresenting the data type, as false !== undefined. I think literally `undefined` would make sense, or to provide a way to clearly indicate it's optional the same way you do with `required`?

File: docs/content/Header.mdx
Comment (colebemis) [reply docs/content/Header.mdx:8]: Perfect ✨

File: docs/src/props-table.js
Comment (colebemis) [reply docs/src/props-table.js:81]: 👍

File: docs/content/Label.mdx
Comment (colebemis) [reply docs/content/Label.mdx:56]: Since `styled-system` is required by the whole library, I don't think I'd consider it an unnecessary dep

**Review by colebemis — APPROVED** [review]
Should be good to go after the merge conflicts are addressed 👍

**Pull request merged**
